src/include/miopen/conv/invokers/impl_gemm_dynamic.hpp

```diff
@@ -26,35 +26,270 @@
+    opArgs.emplace_back(c);
+    opArgs.emplace_back(ho);
+    opArgs.emplace_back(wo);
+    opArgs.emplace_back(stride_h);
+    opArgs.emplace_back(stride_w);
+    opArgs.emplace_back(dilation_h);
+    opArgs.emplace_back(dilation_w);
+    opArgs.emplace_back(pad_h);
+    opArgs.emplace_back(pad_w);
+    opArgs.emplace_back(y);
+    opArgs.emplace_back(x);
+    if(conv_problem.IsFp16())
+    {
+        opArgs.emplace_back(group);
+        opArgs.emplace_back(magic_0);
+        opArgs.emplace_back(magic_1);
+        opArgs.emplace_back(magic_2);
+        opArgs.emplace_back(magic_3);
+        opArgs.emplace_back(magic_4);
+        opArgs.emplace_back(magic_5);
+        opArgs.emplace_back(magic_6);
+        opArgs.emplace_back(shift_pack_0);
+        opArgs.emplace_back(shift_pack_1);
+    }
+    opArgs.emplace_back(pack0);
+
+    return opArgs;
+}
+
+inline float CallImplGemmDynamicForward(const miopen::Handle& handle,
+                                        ConstData_t src,
+                                        Data_t dst,
+                                        ConstData_t wei,
+                                        const std::vector<OpKernelArg>& opShapeArgs,
+                                        const std::vector<KernelInvoke>& kernels)
+{
+    float elapsed = 0.0f;
+    auto kernel   = kernels[0];
+    std::vector<OpKernelArg> opArgs;
+    opArgs.emplace_back(src);
+    opArgs.emplace_back(wei);
+    opArgs.emplace_back(dst);
+
+    opArgs.insert(opArgs.end(), opShapeArgs.begin(), opShapeArgs.end());
+
+    kernel(opArgs);
+
+    if(handle.IsProfilingEnabled())
+        elapsed += handle.GetKernelTime();
+    return elapsed;
+}
+
+template <typename T>
+static inline InvokerFactory MakeImplGemmDynamicForwardInvokerFactory(const ConvolutionContext& ctx,
+                                                                      const T& cfg)
+{
+    const auto& conv_problem = ctx.conv_problem;
+    auto opShapeArgs         = ComputeDynamicIGemmForwardKernelArgs<T>(conv_problem, cfg);
+    return [opShapeArgs](const std::vector<Kernel>& kernels) {
+        return [=](const Handle& handle, const AnyInvokeParams& primitive_parameters) {
+            decltype(auto) data_ctx = primitive_parameters.CastTo<conv::DataInvokeParams>();
+            const auto& tensors     = data_ctx.tensors;
+            auto kernel             = handle.Run(kernels[0]);
+
+            std::vector<KernelInvoke> ks;
+            std::transform(kernels.begin(),
+                           kernels.end(),
+                           std::back_inserter(ks),
+                           [&](const Kernel& k) { return handle.Run(k); });
+            float elapsed = 0;
+            elapsed       = CallImplGemmDynamicForward(
+                handle, tensors.in, tensors.out, tensors.w, opShapeArgs, ks);
+            if(handle.IsProfilingEnabled())
+            {
+                handle.ResetKernelTime();
+                handle.AccumKernelTime(elapsed);
```

Comment:
Manual adjustments of kernel time are not necessary if a Solution consists of only one kernel.

Reply:
If a solution have 2 kernels, do we manually adjust kernel time like `handle.ResetKernelTime(); handle.AccumKernelTime(elapsed);`?

Reply:
Yes, for 2 or more kernels.